tests/test_queue.py

```diff
@@ -0,0 +1,129 @@
+import os
+import random
+
+import logging
+from pathlib import Path
+from typing import List 
+
+import pytest
+
+from psij import Job, JobSpec, JobAttributes, JobExecutor, ResourceSpecV1
+from tempfile import TemporaryDirectory
+
+from executor_test_params import ExecutorTestParams
+from _test_tools import _get_executor_instance, _get_timeout, assert_completed, _make_test_dir
+
+
+SCHEDULER_COMMANDS = {
+    "slurm": {
+        "get_queues": "mdiag -c",
+        "get_user_jobs": "squeue -u $(whoami)",
+        "kill_command": "scancel"
+    },
+    "lsf": {
+        "get_queues": "bqueues -u $(whoami)",
+        "get_user_jobs": "bjobs",
+        "kill_command": "bkill"
+    }
+}
+
+
+def get_slurm_queues() -> List[str]:
+    out = os.popen("mdiag -c").read().split("\n")
+    return [line.split("=")[-1] for line in out if "PartitionName" in line]
+
+
+def get_lsf_queues() -> List[str]:
+    valid_queues = []
+    out = "".join(os.popen("bqueues -u $(whoami) -o 'QUEUE_NAME NJOBS PEND RUN SUSP STATUS'").read()).split("\n")
+    out = [l for l in out if len(l) != 0]
+    out = [l.split(" ") for l in out]
+    if len(out) == 0:
+        return []
+    out.pop(0) # popping headers 
+
+    for queue_info in out: 
+        name = queue_info[0]
+        njobs = int(queue_info[1])
+        pend = int(queue_info[2])
+        run = int(queue_info[3])
+        susp = int(queue_info[4])
+        status = str(queue_info[5])
+
+        if "active" not in status.lower():
+            continue
+
+        if (njobs + pend + run + susp) > 10:
+            valid_queues.append(name)
+
+    return valid_queues
+
+
+def get_queue_info(executor: str) -> List[str]:
+    res = []
+    command = SCHEDULER_COMMANDS[executor]["get_user_jobs"]
+    res.extend(os.popen(command).read().split("\n"))
+    return res
+
+
+def kill_job(scheduler: str, job: Job) -> None:
+    command = f"{SCHEDULER_COMMANDS[scheduler]['kill_command']} {job._native_id}"
+    print("Kill command:", command)
+    os.system(command)
+
+
+def make_job(queue:str) -> Job:
+    return Job(
+        JobSpec(
+            executable="/bin/date",
+            attributes=JobAttributes(
+                queue_name=queue,
+            ),
+        )
+    )
+
+
+def test_queue(execparams: ExecutorTestParams) -> None:
+    scheduler = ""
+    queues = []
+    slurm_queues = get_slurm_queues()
+    lsf_queues = get_lsf_queues()
+
+    queues.extend(slurm_queues)
+    queues.extend(lsf_queues)
+
+    if len(slurm_queues) != 0:
+        scheduler = "slurm"
+    elif len(lsf_queues) != 0:
+        scheduler = "lsf"
```

Comment:
Can you take a look at my implementation? I kept the way I was detecting it, but am now only running the test when `execparams.launcher == "single"`

Reply:
It's not about detecting the LRM on the system but the fact that we test multiple executors on that system. So even if you restrict it to the single launcher, it will still be repeated for the local, batch-test, and whatever PSI/J detected to be the scheduler.

You could remove execparams, but then you risk not having access to other necessary parameters that might be set by the users that set up the tests. By the way, you may want to use execparams.custom_attributes, since some systems require setting various things, like an account or project.